plugin.py

```diff
@@ -29,8 +29,10 @@ def plugin_info(self):
         return plugin_info
 
     def process(self, trace, data_context):
-        with trace.open(data_type='text', mode='wb') as writer:
-            kaitai_utils.write_to_json(trace.open(), writer, kaitai_utils.get_kaitai_class())
+        with trace.open(data_type='text', mode='wb') as writer, trace.open() as data:
+            kaitaiclass = kaitai_utils.get_kaitai_class()
+            json_writer = kaitai_utils.JsonWriter(writer, trace)
```

Comment:
Maybe create a utility method that uses the JsonWriter?
(replace this and the next line with:

```
kaitai_utils.write_to_json(trace, kaitai_utils.get_kaitai_class())
```
(you can handle the open-s in the util class as well then)

Reply:
Done!